hcl2template/parser.go

```diff
@@ -247,9 +257,14 @@ func sniffCoreVersionRequirements(body hcl.Body) ([]VersionConstraint, hcl.Diagn
 	return constraints, diags
 }
 
-func (cfg *PackerConfig) Initialize() hcl.Diagnostics {
+func (cfg *PackerConfig) Initialize(opts packer.InitializeOptions) hcl.Diagnostics {
 	var diags hcl.Diagnostics
 
+	if cfg.DataSources != nil {
+		moreDiags := cfg.evaluateDatasources(opts.ValidationOnly)
+		diags = append(diags, moreDiags...)
+	}
```

Comment:
Here input variables must be usable in data sources, is that the case ?
Also can I use an input var in a local ?

Reply:
Input variables works on data sources. In `PackerConfig.startDatasource` I decode the datasource block with 
`decodeHCL2Spec(body, cfg.EvalContext(nil), datasource)`

Did you mean to use a data source variable in a local? If it so, the answer is yes.